### PR TITLE
add proper additional document update change handler in incremental s…

### PR DESCRIPTION
**Pull request by heejaechang**
…olution updater

unit test added

**Customer scenario**

Customer add or modify additional documents to a C#/VB project then VS crashes.

**Bugs this fixes:** 

https://devdiv.visualstudio.com/web/wi.aspx?pcguid=011b8bdf-6d56-4f87-be0d-0092136884d9&id=370095

**Workarounds, if any**

add/modify additional file manually and then load solution. or turn off out of proc through internal registry key.

**Risk**

areas that have changed are covered by unit tests so I would say risk is low.

**Performance impact**

code change is not related to any area where performance is concerned. 

**Is this a regression from a previous update?**

Yes.

**Root cause analysis:**

unit test missed this area. only covered initial additional file, not modification of the file once solution is loaded. the regression happened while improving performance of updating solution in Roslyn out of proc

**How was the bug found?**

Testing and Dogfooding.

### CODE REVIEW

**Comment by heejaechang** [comment]
@srivatsn @mavasani @panopticoncentral @dotnet/roslyn-ide can you take a look?

File: src/VisualStudio/Core/Test.Next/Services/SolutionServiceTests.cs [hunk 1]
```diff
@@ -191,42 +191,110 @@ public class SolutionServiceTests
...
             using (var workspace = await TestWorkspace.CreateCSharpAsync(code))
             {
-                var map = new Dictionary<Checksum, object>();
+                await VerifySolutionUpdate(workspace, newSolutionGetter);
+            }
+        }
 
-                var solution = workspace.CurrentSolution;
-                var service = await GetSolutionServiceAsync(solution, map);
+        private static async Task VerifySolutionUpdate(TestWorkspace workspace, Func<Solution, Solution> newSolutionGetter)
```
Comment (heejaechang) [inline src/VisualStudio/Core/Test.Next/Services/SolutionServiceTests.cs]: this is just extract method. there is actually no change here. not sure why github think there is change here.

**Comment by heejaechang** [comment]
@aakash-johari can you let us know whether the fix fixed your issue.

**Comment by aakash-johari** [comment]
@heejaechang The fix works for us. Thanks!

**Comment by heejaechang** [comment]
moved bug back to master.

**Comment by heejaechang** [comment]
ping? can I get code review?

**Comment by heejaechang** [comment]
@MattGertz can I get approval for RTW for monday shiproom?

File: src/VisualStudio/Core/Test.Next/Services/SolutionServiceTests.cs (hunk 1 above)
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Test.Next/Services/SolutionServiceTests.cs]: I guess some whitespace changes. Diffing with ?w=1 in the URL looks clean.

**Comment by heejaechang** [comment]
@jasonmalinowski ya, looks like when I changed OOP to be incrementally updated, I forgot to add additional file change tests and missed issues there.

except OOP, I believe additional files are covered by tests pretty well.

**Comment by jasonmalinowski** [comment]
@heejaechang Clearly we had a test hole, so are we trying other exploratory testing here to make sure we got everything for realz?

**Comment by heejaechang** [comment]
@jasonmalinowski yes, we had testing hole with OOP. can't say covered 100% cases. I made OOP to crash right way if there is any inconsistency. so if there is more issue, we will know :) but so far, all cases reported, when tested no longer reproed.

**Comment by natidea** [comment]
@heejaechang please redirect this to RC3

**Pull request merged**